### PR TITLE
[tensor] Apply bcq

### PR DESCRIPTION
**Pull request by lhs8928**
[tensor] apply bcq
    
     - Enable bcq to NNTrainer
    
    Signed-off-by: hyeonseok lee <hs89.lee@samsung.com>

### CODE REVIEW

File: nntrainer/tensor/tensor.cpp [hunk 1]
```diff
@@ -3199,6 +3232,16 @@ void Tensor::read(std::ifstream &file, Tdatatype s_type) {
   NNTR_THROW_IF(!contiguous, std::invalid_argument)
     << getName() << " is not contiguous, cannot read.";
 
+  uint16_t prev_quantized_bit_size = quantized_bit_size;
+
+  if (getDataType() == Tdatatype::BCQ32) {
+    file.read((char *)&quantized_bit_size, sizeof(uint16_t));
+
+    // @todo: reallocate tensor
+    if (prev_quantized_bit_size != quantized_bit_size) {
```
Comment (jihochu) [inline nntrainer/tensor/tensor.cpp:3241]: If it's not implemented yet, plz comment them.

File: nntrainer/tensor/meson.build [hunk 2]
```diff
@@ -52,3 +53,5 @@ endforeach
 foreach h : tensor_headers
   nntrainer_headers += meson.current_source_dir() / h
 endforeach
+
+install_subdir('BiQGEMM', install_dir : nntrainer_prefix / 'include' / 'nntrainer')
```
Comment (SeoHyungjun) [inline nntrainer/tensor/meson.build:57]: No newline at end of file

**Review by SeoHyungjun — APPROVED** [review]

File: nntrainer/tensor/tensor.cpp [hunk 3]
```diff
@@ -3251,10 +3294,36 @@ void Tensor::read(std::ifstream &file, Tdatatype s_type) {
...
+      file.read((char *)&scale, sizeof(__fp16));
+      scale_factors_fp16.push_back(scale);
+    }
   }
 
   checkedRead(file, (char *)getData(), sz, "[Tensor::read] operation failed");
   putData();
+
+  if (getDataType() == Tdatatype::BCQ32) {
+    // BCQ assumtion: width (hidden dimention) is paced by 32 bit
```
Comment (djeong20) [inline nntrainer/tensor/tensor.cpp:3311]: please check typo

File: nntrainer/tensor/tensor.h [hunk 4]
```diff
@@ -1436,10 +1438,19 @@ class Tensor {
...
       return (size() * dim.getDataTypeSize() + 1) / 2;
+    } else if (getDataType() == Tdatatype::BCQ32) {
+      return dim.getDataTypeSize() * quantized_bit_size * dim.height() *
+             ((dim.width() + 31) / 32);
     }
     return size() * dim.getDataTypeSize();
   }
 
+  /**
+   * @brief     Get bcqw
```
Comment (djeong20) [inline nntrainer/tensor/tensor.h:1449]:
```suggestion
   * @brief     Get bcqhw
```

**Pull request closed without merging**